tests/unit/src/main/scala/tests/TestingServer.scala

```diff
@@ -792,6 +792,45 @@ final class TestingServer(
     }
   }
 
+  def assertTypeDefinition(
+      filename: String,
+      query: String,
+      expected: l.Location
+  ): Future[Unit] = {
+    def locationToString(loc: l.Location): String = {
+      val (uri, start, end) =
+        (loc.getUri, loc.getRange.getStart, loc.getRange.getEnd)
+      s"$uri [${start.getLine}:${start.getCharacter} -> ${end.getLine}:${end.getCharacter}]"
+    }
+
+    for {
+      typeDefinitions <- typeDefinition(filename, query)
+    } yield {
+      typeDefinitions.foreach(
+        location =>
+          DiffAssertions.assertNoDiff(
+            locationToString(location),
+            locationToString(expected)
+          )
+      )
+    }
+  }
+
+  def typeDefinition(
+      filename: String,
+      query: String
+  ): Future[List[l.Location]] = {
+    for {
+      (_, params) <- offsetParams(filename, query, workspace)
+      definitions <- server
+        .typeDefinition(params)
+        .asScala
+        .map(l => l.asScala.toList)
```

Comment:
this map can be moved to the `yield` clause which would increase the readability as:
a) the server.typeDefinition(...).asScala chain can be put in a single line
b) the result transformations would be in a single place: `yield { definitions.asScala.toList } `